pkg/plugins/pg/README.md

```diff
@@ -1,144 +1,149 @@
+connector releases it's lock and switches into CDC mode. 
+
+This behavior is enabled by default, but can be turned off by adding 
+`"snapshot":"off"` to the Source configuration.
 
 ## Change Data Capture
-This connector implements CDC features for PostgreSQL by reading WAL log events 
-into a buffer that is checked on each Read request after the initial table Rows 
-have been read.
+This connector implements CDC features for PostgreSQL by reading WAL events 
+into a buffer that is checked on each call of `Read` after the initial snapshot
+has occurred. If there is a record in the buffer, it grabs and returns that 
+record. If it's empty, it returns `ErrEndData` signaling that Conduit should 
+backoff-retry.
 
-## CDC  Configuration
-When Open is called, the connector will attempt to start all necessary 
-connections and will run some initial setup commands to create the logical 
-replication slots it needs to run and start its own subscription. 
+ This behavior is enabled by default, but can be turned off by adding 
+ `"cdc": "off"` to the Source configuration.
 
-The connector user specified in the connection URL must have sufficient 
-privileges to run all of these commands or it will fail.
+### CDC  Configuration
+When the connector switches to CDC mode, it attempts to start all necessary 
+connections and runs the initial setup commands to create its logical 
+replication slots and publications. It will connect to an existing slot if one
+with the configured name exists.
 
-If the `cdc` field in the plugin.Configuration is set to any truthy value, it 
-will be enabled. 
+The Postgres user specified in the connection URL must have sufficient 
+privileges to run all of these setup commands or it will fail.
 
 Publication and slot name are user configurable, and must be correctly set. 
 The plugin will do what it can to be smart about publication and slot 
 management, but it can't handle everything.
 
-If a replication_url is provided, it will be used for CDC features instead of 
-the url. If no replication_url is provided, but cdc is enabled, then it will 
+If a `replication_url` is provided, it will be used for CDC features instead of 
+the url. If no `replication_url` is provided, but cdc is enabled, then it will 
 attempt to use that url value for CDC features and logical replication setup.
-
 Example configuration for CDC features:
 ```
 "cdc":              "true",
 "publication_name": "meroxademo",
 "slot_name":        "meroxademo",
-"url":              url,
+"url":              url, // connection url to the database
 "replication_url":  url,
-"table":            "records",
-"columns":          "key,column1,column2,column3",
+"key":              "key", // postgres column name of your table's primary key
+"table":            "records", // table that the connector should watch
+"columns":          "key,column1,column2,column3", // columns to include in payload
```

Comment:
I wouldn't add comments here if you ask me, this info should go into the "Configuration Options" chapter.